### ISSUE TITLE
nginx

### ISSUE THREAD

**Issue by stephentcannon** [arunoda/meteor-up-legacy#30]
What about nginx as part of the setup with accompanying nginx conf files?

**Comment by iwoork** [comment]
Would love to have this as well infront of the meteor app

**Comment by Multiply** [comment]
There's so many different ways apps work, so how would you have nginx being set up?

I just made a few configs for some of my own projects, where some include SSL, some doesn't. Some needs redirects from other domains, and so on.

If you have a good way to solve it, I wouldn't mind integrating it.

**Comment by arunoda** [comment]
I think that's not a job of mup. Mup will get too complicated if so.

**Comment by Multiply** [comment]
@arunoda I agree. Yet, wouldn't it be smart to be able to store an nginx.conf (or any other type of file, for that matter) file in your own project, and have a way to do symlinks, and restarts of any given service?

For example you have a folder with files, to use for deployment, in which I would (for nginx) store a "nginx.conf" file, a "ssl.crt" and a "ssl.key" file. Then in the `mup.json` file, add variables like

```
{
  "appName": "com.domain",
  "symlink": {
    "nginx.conf": "/etc/nginx/sites-available/com.domain.conf"
  },
  "service": {
    "nginx": "reload"
  }
}
```

And refer to the two SSL-files directly from the nginx.conf and to the SSL-files like so:

```
server {
  # ...
  ssl_certificate     /opt/com.domain/deployment_files/ssl.crt;
  ssl_certificate_key /opt/com.domain/deployment_files/ssl.key;
  # ...
}
```

What should happen, is:
- Includes the "deployment_files" folder (call it something better suited, maybe?), when publishing it to the server
- Make symlinks from target to point, as per the `symlink` object
- Send commands to services, as per the `service` object

Let me know what you think. ;)

Edit:
If you want to deploy to multiple servers, you most likely don't want a service like nginx running on more than 1 server, so I guess my above code, should be defined in the server section, if it should be implemented. Again... This quickly gets out of the scope for mup as I see it.

**Comment by arunoda** [comment]
Now we've the SSL support. Check docs: https://github.com/arunoda/meteor-up#ssl-support
